docs/modules/ROOT/pages/contributors/naming-conventions.adoc

```diff
@@ -0,0 +1,212 @@
+[source,yaml]
+----
+apiServer:
+  # BAD:
+  # deployment: {{ .Release.Name }}-api-server
+  # serviceAccount: {{ .Release.Name}}-api-server
+  # service: {{ .Release.Name }}-api-server
+  # role: {{ .Release.Name }}-api-server
+  # roleBinding: {{ .Release.Name }}-api-server
+  # GOOD:
+  name: {{ .Release.Name }}-api-server
+----
+
+* Do not set the `namespace` metadata field if it's going to have the same
+value as `{{ .Release.Namespace }}`. It's redundant and may introduce an error.
++
+.template-example.yaml
+[source,yaml]
+----
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ .Release.Name }}-test
+  # BAD - namespace: {{ .Release.Namespace }}
+----
+
+* Extract the variables that the user may want to change to the [filename]`values.yaml`
+file.
+If you want to reuse a variable that is not meant to be exposed, go to the next point.
++
+.values.yaml
+[source,yaml]
+----
+apiServer:
+  port: 8081
+----
++
+.template-example.yaml
+[source,yaml]
+----
+kind: Service
+apiVersion: v1
+spec:
+  ports:
+    - protocol: TCP
+      # BAD:
+      # port: 8081
+      # targetPort: 8081
+      # GOOD:
+      port: {{ .Values.apiServer.port }}
+      targetPort: {{ .Values.apiServer.port }}
+      name: server
+----
++
+* Create a variable in the template to reuse it.
+This is the recommended approach for variables that are not meant
+to be exposed to the user.
++
+.template-example.yaml
+[source,yaml]
+----
+{{- $caName := tpl "{{ .Release.Name }}-ca" . -}}
+---
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name: "{{ $caName }}"
+spec:
+  selfSigned: {}
+---
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name: "{{ .Release.Name }}"
+spec:
+  ca:
+    secretName: "{{ $caName }}"
+----
+
+* Take advantage of the
+link:[tpl function]
+for variables exposed in [filename]`values.yaml` that contains a repeated part
+extracted in another variable.
+
++
+.values.yaml
+[source,yaml]
+----
+sink:
+  name: {{ .Release.Name }}-sink
+----
++
+.template-example.yaml
+[source,yaml]
+----
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ tpl .Values.sink.name . }}
+----
++
+[NOTE]
+====
+The tpl function also allows nested templating:
+
+.values.yaml
+[source,yaml]
+----
+database:
+  name: "{{ .Release.Name }}-database" # <1>
+  url: "{{ tpl .Values.database.name . }}.{{ .Release.Namespace }}.svc.cluster.local" # <2>
+----
+.template-example.yaml
+[source,yaml]
+----
+apiVersion: apps/v1
+kind: Deployment
+spec:
+  template:
+    spec:
+      containers:
+        - env:
+          - name: POSTGRES_URL
+            value: {{ tpl .Values.database.url . }} # <3>
+----
+<1> The database name include the release name as a prefix, so the variable is templated.
+<2> The URL contain the database name, as this is a templated variable we need to use tpl
+to compile its value on runtime.
+<3> We need a second usage (nested) of tpl function to extract the value of the url value into
+the template, as it is not a static value.
```

Comment:
```suggestion
<1> The database name includes the release name as a prefix, so the variable is templated.
<2> The URL contains the database name, since this is a templated variable we need to use tpl
to compile its value at runtime.
<3> We need a second usage (nested) of the tpl function to extract the value of the url into
the template, as it is not a static value.
```

Reply:
Fix applied